code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -148,4 +159,59 @@ public void error(ExtensionError extensionError) {
             }
         });
     }
+
+    /**
+     * Sets the notification factory which will be used by {@link MessagingPushNotificationHandler} to create the push notification
+     * @param factory
+     */
+    public static void setPushNotificationFactory(IMessagingPushNotificationFactory factory) {
+        if (factory == null) {
+            // log
+            return;
+        }
+        notificationFactory = factory;
+    }
+
+    /**
+     * Sets the image downloader which will be used for downloading large icon and images for notification
+     * @param downloader
+     */
+    public static void setPushImageDownloader(IMessagingImageDownloader downloader) {
+        if (downloader == null) {
+            // log
+            return;
+        }
+        imageDownloader = downloader;
+    }
+
+    public static boolean handlePushNotificationWithRemoteMessage(final RemoteMessage remoteMessage) {
+        if (remoteMessage == null) {
+            // Log remote message is null
+            return false;
+        }
+
+        final MessagingPushPayload payload = new MessagingPushPayload(remoteMessage);
+
+        if (!payload.isAEPPushMessage()) {
+            // Log not aep message
+            return false;
+        }
+
+        final Context appContext = MobileCore.getApplication().getApplicationContext();
+        final int notificationId = (int)System.currentTimeMillis();
+        final String messageId = remoteMessage.getMessageId();
+
+        new Thread(new Runnable() {
```

Comment:
What do you think about using a thread pool with single thread instead of creating a new thread every time?